File: src/p0056/python/solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        temp = sorted(intervals)
        ans = [temp[0]]
        for i in range(1, len(temp)):
            if ans[-1][1] < temp[i][0]:
                ans.append(temp[i])
            elif ans[-1][1] >= temp[i][0] and ans[-1][1] <= temp[i][1]:
                ans[-1][1] = temp[i][1]
                
        return ans

File: src/p0056/golang/solution.go
func merge(intervals [][]int) [][]int {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})
	ans := [][]int{intervals[0]}
	for i := 1; i < len(intervals); i++ {
		if ans[len(ans)-1][1] >= intervals[i][1] {
			continue
		}
		if ans[len(ans)-1][1] >= intervals[i][0] {
			ans[len(ans)-1][1] = intervals[i][1]
		} else if ans[len(ans)-1][1] < intervals[i][0] {
			ans = append(ans, intervals[i])
		}
	}

	return ans
}

// 华为